crypto/s2n_ecc_evp.c

```diff
@@ -166,6 +175,14 @@ static int s2n_ecc_evp_generate_key_nist_curves(const struct s2n_ecc_named_curve
     return 0;
 }
 
+static int s2n_ecc_evp_generate_key_noop(const struct s2n_ecc_named_curve *named_curve, EVP_PKEY **evp_pkey)
+{
+    /* We should never hit this function — pure ML-KEM doesn’t generate an EC keypair.
+    * The placeholder curve only acts as a filter, so if this ever runs it indicates a bug.
+    */
```

Comment:
nit: This noop function isn't really specific to the pure pq placeholder anymore. Probably best to make this more generic or just remove it. Might be clear enough from the error that this should never be called anyway.
```suggestion
```